src/features/wallet/context/FuelWalletContext.tsx

```diff
@@ -1,33 +1,68 @@
 'use client';
 
-import { FuelProvider, useWallet } from '@fuel-wallet/react';
-import { PropsWithChildren, useEffect } from 'react';
+import { defaultConnectors } from '@fuels/connectors';
+import { FuelProvider, NetworkConfig, useNetwork, useWallet } from '@fuels/react';
+import { PropsWithChildren, useEffect, useState } from 'react';
 
+import { Provider, TESTNET_NETWORK_URL, Wallet, WalletLocked } from 'fuels';
 import { logger } from '../../../utils/logger';
-import { reinitializeWarpCore } from '../../store';
+import { reinitializeWarpCore, useStore } from '../../store';
+
+const FUEL_MAINNET_NETWORK = { chainId: 9889, url: 'https://mainnet.fuel.network/v1/graphql' };
+const FUEL_TESTNET_NETWORK = { chainId: 0, url: TESTNET_NETWORK_URL };
 
 function FuelWalletTracker() {
   const { wallet } = useWallet();
+  const { network } = useNetwork();
 
   useEffect(() => {
     async function updateWalletInWarpCore() {
       try {
-        if (!wallet) return;
-        logger.info('Reinitializing WarpCore with fuelwallet access');
-        await reinitializeWarpCore(wallet);
+        if (!wallet || !network) return;
+
+        logger.warn('Reinitializing WarpCore with fuelwallet access', { network: network?.url });
+        const provider = new Provider(network.url);
+        const lockedWallet = Wallet.fromAddress(
+          wallet.address,
+          provider,
+        ) as unknown as WalletLocked;
+
+        await reinitializeWarpCore(lockedWallet);
       } catch (error) {
         logger.error('Error in updateWalletInWarpCore:', error);
       }
     }
     updateWalletInWarpCore();
-  }, [wallet]);
+  }, [wallet, network]);
 
   return null;
 }
 
 export default function FuelWalletContext({ children }: PropsWithChildren<unknown>) {
+  const { transferLoading } = useStore();
+  const currentState = useStore.getState();
+  const [networks, setNetworks] = useState<NetworkConfig[]>([FUEL_MAINNET_NETWORK]);
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
```

Comment:
What happens if the fuel wallet is not installed?

Reply:
I think the user will attempt to connect but they won't get any error. I tried it out on incognito.